### ISSUE TITLE
Crash in CellularService.currentRadioAccessTechnology

### ISSUE THREAD

**Issue by stevenp** [mopub/mopub-ios-sdk#377]
I'm seeing the crash pasted below in Crashlytics. I've only seen one report so far, but I figured the report might be helpful.

#### MoPub SDK Version:
5.16.1

#### Device model and OS Version:
iPhone 11, iOS 14.0.1

#### Evidence:
Crashed: com.apple.main-thread
0  RoutesyFree                    0x1032cee54 CellularService.currentRadioAccessTechnology.getter + 71 (CellularService.swift:71)
1  RoutesyFree                    0x1032d3be0 static DeviceInformation.currentNetworkStatus.getter + 103 (DeviceInformation.swift:103)
2  RoutesyFree                    0x1032d3adc @objc static DeviceInformation.currentNetworkStatus.getter + 4336532188 (<compiler-generated>:4336532188)
3  RoutesyFree                    0x10324e468 +[MPAdServerURLBuilder(Ad) connectionTypeValue] + 309 (MPAdServerURLBuilder.m:309)
4  RoutesyFree                    0x10324d99c +[MPAdServerURLBuilder(Ad) URLWithAdUnitID:targeting:desiredAssets:adSequence:] + 247 (MPAdServerURLBuilder.m:247)
5  RoutesyFree                    0x10324d7d8 +[MPAdServerURLBuilder(Ad) URLWithAdUnitID:targeting:desiredAssets:] + 230 (MPAdServerURLBuilder.m:230)
6  RoutesyFree                    0x10324d75c (Missing)
7  RoutesyFree                    0x10325644c -[MPBannerAdManager loadAdWithURL:] + 195 (MPBannerAdManager.m:195)
8  RoutesyFree                    0x103255f88 -[MPBannerAdManager loadAdWithTargeting:] + 135 (MPBannerAdManager.m:135)
9  RoutesyFree                    0x103250090 -[MPAdView loadAd] + 133 (MPAdView.m:133)
10 RoutesyFree                    0x102d31614 (Missing)
11 RoutesyFree                    0x102d3340c partial apply for closure #1 in BannerLoadOperation.main() + 4330632204 (<compiler-generated>:4330632204)
12 RoutesyFree                    0x102d34ea8 thunk for @escaping @callee_guaranteed () -> () + 4330639016 (<compiler-generated>:4330639016)
13 libdispatch.dylib              0x18f916fd0 _dispatch_call_block_and_release + 32
14 libdispatch.dylib              0x18f918ac8 _dispatch_client_callout + 20
15 libdispatch.dylib              0x18f92604c _dispatch_main_queue_callback_4CF + 836
16 CoreFoundation                 0x18fc971e4 __CFRUNLOOP_IS_SERVICING_THE_MAIN_DISPATCH_QUEUE__ + 16
17 CoreFoundation                 0x18fc913b4 __CFRunLoopRun + 2508
18 CoreFoundation                 0x18fc904bc CFRunLoopRunSpecific + 600
19 GraphicsServices               0x1a6715820 GSEventRunModal + 164
20 UIKitCore                      0x192634734 -[UIApplication _run] + 1072
21 UIKitCore                      0x192639e10 UIApplicationMain + 168
22 RoutesyFree                    0x102ceec20 main + 34 (AppDelegate.swift:34)
23 libdyld.dylib                  0x18f957e60 start + 4

**Comment by arusu0629** [comment]
@stevenp 
I'm in the same situation.
What is the version of Xcode you are using?

I'm using 12.3.

**Comment by hmertkahraman** [comment]
Hi @stevenp ,

Thanks for sharing us the details. Can we ask what version of XCode do you use to compile this project? Do you happen to have any repro steps available? Is it upon shortly after launching the app?

Thanks,
MoPub Team

**Comment by stevenp** [comment]
@hmertkahraman My app was compiled with Xcode 12.4, and unfortunately I don't have any repro steps because this is a crash report from a user device that I received using Crashlytics. I load my ads almost immediately after the app launches, so I'm assuming that's probably when the crash happens. Out of thousands of sessions, I've seen the crash reported only 7 times, so it seems like it's pretty rare. All of the users are on iOS 14.0.0 or iOS 14.0.1.

**Comment by mmdock** [comment]
i believe this is the same as https://github.com/mopub/mopub-ios-sdk/issues/372

**Comment by HyunjoonKo** [comment]
I'm having the same problem too.
- MoPub iOS SDK 5.16.1
- Built with Xcode 12.4.
- 100% iOS 14 or higher devices.
- Most errors occur on iPhone XS/XR or higher devices.

STACK TRACE:
```
Crashed: com.apple.main-thread
0  MoPubSDK                       0x108583d0c $s8MoPubSDK15CellularServiceC28currentRadioAccessTechnologyAA13NetworkStatusOvg + 1116
1  MoPubSDK                       0x108588dd8 $s8MoPubSDK17DeviceInformationC20currentNetworkStatusAA0gH0OvgZ + 248
2  MoPubSDK                       0x108588cd4 $s8MoPubSDK17DeviceInformationC20currentNetworkStatusAA0gH0OvgZTo + 24
3  MoPubSDK                       0x10850256c +[MPAdServerURLBuilder(Ad) connectionTypeValue] + 44
4  MoPubSDK                       0x108501aa0 +[MPAdServerURLBuilder(Ad) URLWithAdUnitID:targeting:desiredAssets:adSequence:] + 404
5  MoPubSDK                       0x1085018dc +[MPAdServerURLBuilder(Ad) URLWithAdUnitID:targeting:desiredAssets:] + 84
6  MoPubSDK                       0x108501860 +[MPAdServerURLBuilder(Ad) URLWithAdUnitID:targeting:] + 64
7  MoPubSDK                       0x1085430cc -[MPRewardedAdManager loadRewardedAdWithCustomerId:targeting:] + 324
8  MoPubSDK                       0x108545c08 +[MPRewardedAds loadRewardedAdWithAdUnitID:keywords:userDataKeywords:customerId:mediationSettings:localExtras:] + 636
9  Mubeat                         0x105b05edc closure #1 in closure #1 in MoPubManager.request(reward:completion:) + 326 (MoPubManager.swift:326)
10 Mubeat                         0x1052f158c thunk for @escaping @callee_guaranteed () -> () + 4381480332 (<compiler-generated>:4381480332)
11 libdispatch.dylib              0x1aaadafd0 _dispatch_call_block_and_release + 32
12 libdispatch.dylib              0x1aaadcac8 _dispatch_client_callout + 20
13 libdispatch.dylib              0x1aaaea04c _dispatch_main_queue_callback_4CF + 836
14 CoreFoundation                 0x1aae5b1e4 __CFRUNLOOP_IS_SERVICING_THE_MAIN_DISPATCH_QUEUE__ + 16
15 CoreFoundation                 0x1aae553b4 __CFRunLoopRun + 2508
16 CoreFoundation                 0x1aae544bc CFRunLoopRunSpecific + 600
17 GraphicsServices               0x1c18d9820 GSEventRunModal + 164
18 UIKitCore                      0x1ad7f8734 -[UIApplication _run] + 1072
19 UIKitCore                      0x1ad7fde10 UIApplicationMain + 168
20 Mubeat                         0x104ebba74 main + 14 (AppDelegate.swift:14)
21 libdyld.dylib                  0x1aab1be60 start + 4
```

**Comment by hmertkahraman** [comment]
Hi,

Thanks for providing these information. We are currently investigating this issue further and will provide updates soon.

_Side note that issue #372 is related with compile time errors while this one is at runtime so we're trying to identify the exact reasons._ 

Thanks,
Mert

**Comment by hmertkahraman** [comment]
Hi @stevenp ,

This issue is under our radar. We will release a fix in the next release (ETA is currently TBD).

Thanks,
Mert

**Comment by kaanunlu2** [comment]
Hi @hmertkahraman,

Does the new release 5.16.2 fix this issue?

Thanks,
Kaan

**Comment by hmertkahraman** [comment]
Hi,

The necessary fix was shipped on 5.16.2 release, please update and test to ensure. 

Thanks,
Mert
